### PR TITLE
{tools} Added git as a dependency to GitPython version 3.x

### LINKED ISSUE #11901: gitpython does not have git as a pre-requisite and resorting to system install
The GitPython 3.1.9 does not declare git as a dependency.  so it failed on a machine where we hadn’t yum installed git.

### PR DESCRIPTION
**Pull request by jhein32**
fixes #11901

### CODE REVIEW

File: easybuild/easyconfigs/g/GitPython/GitPython-3.1.0-GCCcore-8.3.0-Python-3.7.4.eb [hunk 1]
```diff
@@ -13,6 +13,7 @@ builddependencies = [('binutils', '2.32')]
 
 dependencies = [
     ('Python', '3.7.4'),
+    ('git', '2.23.0'),
```
Comment (boegel) [inline easybuild/easyconfigs/g/GitPython/GitPython-3.1.0-GCCcore-8.3.0-Python-3.7.4.eb:16]: @jhein32 Please use the `-nodocs` variant here too (which should fix the failing tests)

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/g/GitPython/GitPython-3.1.0-GCCcore-8.3.0-Python-3.7.4.eb (hunk 1 above)
Comment (jhein32) [reply easybuild/easyconfigs/g/GitPython/GitPython-3.1.0-GCCcore-8.3.0-Python-3.7.4.eb]: Ok, that worked, but I have a problem with:

```sanity_pip_check = True```

If I add this line, it complains:

```AssertionError: There should be no redefined easyconfig parameters, found 1: sanity_pip_check (2)```

If I remove it, it complains:

```AssertionError: sanity_pip_check should be enabled in GitPython-3.0.3-GCCcore-8.2.0-Python-3.7.2.eb```

Please advise.  Thanks.

Comment (Flamefire) [reply easybuild/easyconfigs/g/GitPython/GitPython-3.1.0-GCCcore-8.3.0-Python-3.7.4.eb]: It is complaining about different ECs. Open the ECs side-by-side and observe that GitPython-3.0.3-GCCcore-8.2.0-Python-3.7.2.eb does not have a `sanity_pip_check = True` line but the others do. You added it to all 3 ECs and removed it from all 3 ECs, so given that 2 out of 3 were correct already you broke either 2 or left 1 broken.

-> Add the line to GitPython-3.0.3-GCCcore-8.2.0-Python-3.7.2.eb at the same place as there is in the other 2, that is above `exts_default_options` and while you are at it add both this lines (only to that EC):

```
use_pip = True
sanity_pip_check = True
```

**Review by Flamefire — APPROVED** [review]

**Comment by lexming** [comment]
@jhein32 I added the `https` to the homepage URLs of `CESM-deps` easyconfigs given that it's a very minor fix. CI tests should be ok for this PR now.

**Comment by jhein32** [comment]
@lexming Thanks.  

So as far as I am concerned this is now ready to roll.  Checks are finally passing.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 9 out of 9 (8 easyconfigs in total)
node381.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/7a001fb293476c12e0d0e00dba7187fe for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 10 out of 10 (8 easyconfigs in total)
node113.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/5126195c1dc8997a2ad6cd40628201c8 for a full test report.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Going in, thanks @jhein32 !

**Pull request merged**